LionOnTheWeb/lionontheweb-web
# Remove and replace gh action for precommit tasks

To be merged to main #52 branch

----

## Changes overview
* Restructured Husky pre-commit functionality:
    * Added color variables to use in subscripts
    * Removed code validation and format scripts for staged files in favor of scripts defined in `package.json`
    * Added branch validation script with a getter utility subscript for extracting branch information
    * Added script to deploy Storybook changes to Chromatic
    * (Local) Created new node env files for env variables:
        * `.env.development.local`
        * `env.production.local`
    * Utilized boolean variable to determine Storybook deployment (currently only on `release/*` branch types)
* Added husky, stylelint and lint-stage dependencies for additional linting utilities used for pre-commit tasks
* Organized and renamed `package.json` scripts for verbose references

## Testing steps
1. Pull down branch
2. Run `yarn` to install the latest dependencies
3. Run `git commit` to make a test commit and verify the pre-commit tasks run

## Screenshots

***Precommit tasks running on dev branch type:***
![Screenshot 2023-05-31 at 7 40 02 PM](https://github.com/LionOnTheWeb/lionontheweb-web/assets/38708680/882406c5-7c0f-443c-8b6c-71592731027f)

***Precommit tasks running on release branch type:***
![Screenshot 2023-05-31 at 7 41 32 PM](https://github.com/LionOnTheWeb/lionontheweb-web/assets/38708680/ebb1874e-7010-4aa0-bda2-99b5d99db8fe)
